src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  static DAMAGE = 50;
+
+  bite(prey) {
+    if (prey instanceof Carnivore) {
+      return;
+    }
```

Comment:
The check `if (prey instanceof Carnivore)` is correct for preventing carnivores from biting each other. However, it might be more intuitive to explicitly check if the prey is a `Herbivore` instead, as the task likely involves interactions between herbivores and carnivores.